src/scalar/dusk.rs

```diff
@@ -262,34 +261,34 @@ impl Scalar {
         res
     }
 
-    /// Creates a `Scalar` from arbitrary bytes by hashing the input with BLAKE2b into a 256-bits
-    /// number, and then converting it into its `Scalar` representation.
-    pub fn from_var_bytes(input: &[u8]) -> Scalar {
+    /// Creates a `Scalar` from arbitrary bytes by hashing the input with
+    /// BLAKE2b into a 512-bits number, and then converting the number into its
+    /// `Scalar` representation by reducing it by the modulo.
+    ///
+    /// This implementation follows the first conversion of
+    /// https://hackmd.io/zV6qe1_oSU-kYU6Tt7pO7Q with concrete numbers:
+    /// ```text
+    /// p = 0x73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001
+    /// p = 52435875175126190479447740508185965837690552500527637822603658699938581184513
+    ///
+    /// s = 2^512
+    /// s = 13407807929942597099574024998205846127479365820592393377723561443721764030073546976801874298166903427690031858186486050853753882811946569946433649006084096
+    ///
+    /// r = 255699135089535202043525422716183576215815630510683217819334674386498370757523
+    ///
+    /// m = 3294906474794265442129797520630710739278575682199800681788903916070560242797
+    /// ```
+    pub fn from_blake2b(input: &[u8]) -> Scalar {
```

Comment:
Isn't the `blake2b` an implementation detail? Meaning: the hasher is irrelevant for the consumer, nor we take it concretely from the arguments

If implementation detail, then it shouldn't leak on the function name as changing the primitive later on would imply hard breaking changes

Reply:
To me `from_var_bytes` didn't convert the meaning that we are essentially hashing to a scalar, so I would like to avoid that name.
Another possibility would be something like `from_hash`.
I thought that specifying which hash might be useful downstream since different hash functions can have different security implications.
Additionally we would keep the option to easily add the same functionality for different hash functions. Whether we will need that is debatable though.

Reply:
The problem with using `from_hash` is we don't take a hash as argument. `from_slice` maybe? But then we still need to suggest it won't perform a canonical conversion from an expected format. Rather, we will use arbitrary bytes and normalize them

Reply:
Hmm not sure, it still implies we are taking something already hashed. Maybe just fallback to your reply ~didn't convert the meaning that we are essentially hashing to a scalar, so I would like to avoid that name~ and go for `hash_to_scalar`

Reply:
That's the one! Changed it